ConnectionFactory.php

```diff
@@ -102,7 +108,11 @@ public function createConnection(array $params, ?Configuration $config = null, ?
             $connection = DriverManager::getConnection($params, $config, $eventManager);
             $params     = $this->addDatabaseSuffix(array_merge($connection->getParams(), $overriddenOptions));
             $driver     = $connection->getDriver();
-            $platform   = $driver->getDatabasePlatform();
```

Comment:
so previously this skipped any server version and for mysql for example just returned `MySQLPlatform`.

We could use `$connection->getDatabasePlatform()` but it would mean we auto-detect the server version which could mean we actually connect to the database. We should avoid that here I guess.

Is there any other way to do this with DBAL 4 except for using a `StaticServerVersionProvider` with a fallback to an empty server version? :see_no_evil:  :thinking: 

@derrabus maybe you have an idea

Reply:
> so previously this skipped any server version and for mysql for example just returned `MySQLPlatform`.

Why though? For MySQL drivers especially, this sounds like a bad idea because we need the server version to determine whether we talk to a MySQL or a MariaDB.

Reply:
We check if its a "Mysql like" platform (`AbstractMySQLPlatform`) to set the `collation` option :thinking: 

See https://github.com/doctrine/DoctrineBundle/blob/b1ff93403daacc835a653db39431f2a492fa87b6/ConnectionFactory.php#L105-L127

Maybe this isn't even needed anymore on DBAL 4? See https://github.com/doctrine/dbal/issues/5214, https://github.com/doctrine/dbal/pull/5246 

I believe `AbstractMySQLPlatform` only handles `collation` on 4.0, or?

Reply:
Hm but we also set a default charset `utf8mb4` for those platforms :thinking: We have to keep this I think?

Reply:
I tried this but it would mean we might connect to the database to determine the platform version (if not specified via config). For example all over the testsuite we would try to connect to some mysql db.

Since this is the initial connection that is replaced further down with changed parameters: we should avoid connecting, or?